doc/source/cluster/kubernetes/user-guides/config.md

```diff
@@ -128,12 +128,27 @@ Therefore, make sure to specify any sidecar containers
 in the `containers` list.
 
 #### resources
-It’s important to specify container CPU and memory requests and limits for
-each group spec. For GPU workloads, you may also wish to specify GPU
-limits. For example, set `nvidia.com/gpu:2` if using an Nvidia GPU device plugin
+It's important to specify container CPU and memory resources for each group spec. Since CPU is a
+[compressible resource], you may want to set only CPU requests and not limits to guarantee your
+workloads a minimum amount of CPU but [allow them to take advantage of unused CPU and not get
+throttled][1] if they use more than their requested CPU.
+
+Memory is not a compressible resource when using Kubernetes with Linux cgroupv1 but is a
+compressible resource with cgroupv2. If you're using cgroupv1, set both memory requests and limits.
+If you're using cgroupv2, you can omit memory limits with the same rationale for CPU.
```

Comment:
@andrewsykim, lmk what you hear back about this from your colleagues.

Reply:
lmk if you heard anything yet, thanks again

Reply:
@davidxia, do you have any references for the cgroup v2 related statements? I can read them so that we can move this PR forward.

Reply:
[Quality-of-Service for Memory Resources](https://kubernetes.io/blog/2021/11/26/qos-memory-resources/#:~:text=Fortunately%2C%20cgroup%20v2,memory%20as%20well.) suggests that with cgroupv2, memory is a compressible resource that can be throttled.

> Fortunately, cgroup v2 brings a new design and implementation to achieve full protection on memory. The new feature relies on cgroups v2 which most current operating system releases for Linux already provide. With this experimental feature, [quality-of-service for pods and containers](https://kubernetes.io/docs/tasks/configure-pod-container/quality-service-pod/) extends to cover not just CPU time but memory as well.

and see following section [How does it work?](https://kubernetes.io/blog/2021/11/26/qos-memory-resources/#how-does-it-work).

Reply:
I tried to look into it. From what I understand, memory is still not compressible in cgroup v2.

> With this experimental feature, [quality-of-service for pods and containers](https://kubernetes.io/docs/tasks/configure-pod-container/quality-service-pod/) extends to cover not just CPU time but memory as well

My current understanding is that cgroup v1 only manages anonymous memory. Therefore, it's still possible for a Pod to use more memory than its limit if it relies on file-backed memory. Therefore, it can't promise QoS for memory.

WIth cgroup v2, it manages both anonymous memory and file-backed memory, so a Pod will no longer use memory more than memory limit.

Reply:
Hm, I'm trying to understand this myself. The page says a resource is compressible if Kubernetes throttles the corresponding container instead of terminating it when it exceeds its resource limits. Later on this part suggests memory is compressible.

> `memory.high` is the memory usage throttle limit. This is the main mechanism to control a cgroup's memory use. If a cgroup's memory use goes over the high boundary specified here, the cgroup’s processes are throttled and put under heavy reclaim pressure.

But I'm not certain. I can remove this paragraph for now if you'd like.

Reply:
I just found that the KEP is stalled.

https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/

<img width="705" alt="image" src="https://github.com/user-attachments/assets/c9e9d4a2-e1ae-4b3e-bc7a-84c86c18aee4" />

Reply:
nice find. I'll remove this paragraph for now.